subsys/bluetooth/shell/bt.c

```diff
@@ -2048,6 +2048,11 @@ static const struct shell_cmd bt_commands[] = {
 	{ "advx", cmd_advx, "<on off> [coded] [anon] [txp]" },
 	{ "scanx", cmd_scanx, "<on passive off> [coded]" },
 #endif /* CONFIG_BT_CTLR_ADV_EXT */
+#if defined(CONFIG_BT_CTLR_DTM)
+	{ "test_tx", cmd_test_tx, "<chan> <len> <type> <phy>" },
+	{ "test_rx", cmd_test_rx, "<chan> <phy> <mod_idx>" },
+	{ "test_end", cmd_test_end, HELP_NONE},
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
```

Comment:
I wonder if split builds, host only, could take advantage of these commands? Or perhaps this should only be limited to the controller since there is no point on testing this from the host?

Reply:
These are not over HCI, but your are right, it should use HCI... will make changes.

Reply:
@Vudentz @cvinayak @jhedberg to use HCI commands we would need new public APIs that enable and disable the DTM, because we don't send HCI commands directly from the shell, we always use APIs? or is sending commands directly from the shell code a possibility?

Reply:
We don't expose raw HCI access as a public API (mainly to cater for architectures that don't have HCI, like we did with NBLE). Since this patch only touches the controller shell, I think it's fine to make direct calls. However, since these are anyway internal APIs that you're calling from the shell, if it's not too messy you could in principle go ahead with the HCI option as well and call bt_hci_cmd_send_sync() etc. A benefit of implementing the HCI commands in the controller would be that we can then consider the split controller/host case and expose these using a public API on the host side.

Reply:
The HCI commands are already implemented in the controller. This is about calling bt_hci_cmd_send_sync() from the test shell, whether it is a good idea or not.